bentoml/server/instruments.py

```diff
@@ -2,13 +2,19 @@
 import multiprocessing
 import os
 import shutil
-from dependency_injector.wiring import Provide, inject
-from flask import Request
 from timeit import default_timer
+from typing import TYPE_CHECKING
+
+from flask import Request
+from simple_di import Provide, inject
 
 from bentoml.configuration.containers import BentoMLContainer
 
 
+if TYPE_CHECKING:
+    Lock = multiprocessing.synchronize.Lock
```

Comment:
Are we introduce forward referencing to save import time? How much time does this save?

Reply:
No, we avoid breaking during running. This attribute only exists when doing type checking, comes with https://github.com/python/typeshed